src/tests/unit_tests/test_vectorstores.py

```diff
@@ -0,0 +1,186 @@
+    """
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Add vectors to Firestore
+    ids = firestore_store.add_texts(
+        ["test_doc_1", "test_doc_2"],
+    )
+
+    # Verify that the first document was added to Firestore
+    # with a generated ID that matches the returned ID
+    doc = firestore_store.collection.document(ids[0]).get()
+    data = doc.to_dict()
+    assert doc.id == ids[0]
+    assert data["content"] == "test_doc_1"
+
+    # Verify that the second document was added to Firestore
+    # with a generated ID that matches the returned ID
+    doc = firestore_store.collection.document(ids[1]).get()
+    data = doc.to_dict()
+    assert doc.id == ids[1]
+    assert data["content"] == "test_doc_2"
+
+
+def test_firestore_add_vectors_assertions():
+    """
+    Tests assertions in FirestoreVectorStore.add_vectors method.
+    """
+
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Test assertions
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts(["test_doc1", "test_doc2"], ids=["1", "2", "3"])
+
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts(
+            ["test_doc1", "test_doc2"], metadatas=[{"foo": "bar"}]
+        )
+
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts(["test_doc1", "test_doc2"], ids=["1"])
+
+    with pytest.raises(AssertionError):
+        firestore_store.add_texts([])
+
+
+def test_firestore_update_vectors():
+    """
+    An end-to-end test for updating vectors in FirestoreVectorStore.
+    """
+
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Add vectors to Firestore
+    firestore_store.add_texts(["test1", "test2"], ids=["1", "2"])
+
+    # Verify that the vectors were updated in Firestore
+    docs = firestore_store.collection.stream()
+    for doc in docs:
+        data = doc.to_dict()
+        if doc.id == "1":
+            assert data["content"] == "test1"
+            assert data["metadata"] == {"foo": "bar"}
+        elif doc.id == "2":
+            assert data["content"] == "test2"
+            assert data["metadata"] == {"baz": "qux"}
+
+
+def test_firestore_similarity_search():
+    """
+    An end-to-end test for similarity search in FirestoreVectorStore.
+    """
+
+    # Create FirestoreVectorStore instance
+    firestore_store = FirestoreVectorStore(
+        TEST_COLLECTION, get_embeddings(), client=get_client()
+    )
+
+    # Add vectors to Firestore
+    firestore_store.add_texts(["test1", "test2"], ids=["1", "2"])
+
+    # Perform similarity search
+    results = firestore_store.similarity_search("test1", k=2)
```

Comment:
The signature of this method is a List ,  why are we iterating below to aggregate instead of using just `len()`?